### PR TITLE
change bamm to samm

### LINKED ISSUE #744: [Model Issue]: bamm instead of samm in bom_as_specified
## URL to Model with Issue
https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.bom_as_specified/1.0.1/BomAsSpecified.ttl

## Issue Description 
The model above references only BAMM and not SAMM.
If I understand correctly, BAMM has been renamed / replaces by SAMM. In that case, the model needs to be corrected.

### PR DESCRIPTION
**Pull request by gautam-e**
## Description
This addresses #744. `bamm` has been renamed to `samm` and the version has been bumped (patch).

 -->

Closes #744 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.6.0)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [ ] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [ ] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [ ] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [ ] All required reviewers have approved this PR (see reviewers section)
- [ ] The new aspect (version) will be implemented by at least one data provider
- [ ] The new aspect (version) will be consumed by at least one data consumer
- [ ] There exists valid test data
- [ ] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [ ] The model has at least version '1.0.0'
- [ ] If a previous model exists, model deprecation has been checked for previous model
- [ ] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by agg3fe** [comment]
@gautam-e The version for this model should be major update, means it should be 3.0.0 (as 2.0.0 was already removed).
Also, please update io.catenax.bom_as_specified/RELEASE_NOTES.md for this new model.

**Comment by gautam-e** [comment]
@agg3fe changes done. Please check if ok.

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+:partQuantity a samm:Property;
+    samm:preferredName "part quantity"@en;
+    samm:description "This is the quantity how often this part is in the item."@en;
+    samm:characteristic :PartQuantityCharacteristic .
+
+:partDescription a samm:Property;
+    samm:preferredName "part description"@en;
+    samm:description "The description of the part in the primary language of the production facility of the product owner."@en;
+    samm:characteristic samm-c:Text .
```
Comment (jSchuetz88) [inline io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:143]: No description provided for simple type.

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl [hunk 2]
```diff
@@ -0,0 +1,197 @@
+#######################################################################
+# Copyright (c) 2022 T-Systems International GmbH
+# Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+# Copyright (c) 2022 Volkswagen AG
+# Copyright (c) 2022 Robert Bosch GmbH
+# Copyright (c) 2022 ZF Friedrichshafen AG
+# Copyright (c) 2022 Contributors to the Eclipse Foundation
```
Comment (jSchuetz88) [inline io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:7]: The copyrights should be updated.

**Review by jSchuetz88 — CHANGES_REQUESTED** [review]
Thank you for updating the BAMM version to SAMM. From my side only minor changes are requested regarding:

- [ ] all model elements at least contain the fields "preferred name" and "description" in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [ ] preferredName should be human readable and follow normal orthography (e.g., no camel case but normal word separation)
- [ ] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    samm:exampleValue "abcA76" .
+
+:item a samm:Property;
+    samm:preferredName "item"@en;
+    samm:description "Top-level separation of parts which represent a set and shall indicate that the individual items are within each other completely assembled and treated as a unit, but are separable units, e.g. the \"vehicle\", \"electricCharger\" and the \"winter wheels\" are separated and could over lifetime be sold as individual items by the end customer.\n \nFor other end-customer products a set might be like \"cordless screwdriver\" and 2 times \"supplemental battery\"."@en;
+    samm:characteristic :ItemCharacteristic .
+
+:IdentifierCharacteristic a samm:Characteristic;
+    samm:preferredName "IdentifierCharacteristic"@en;
+    samm:dataType xsd:string .
```
[the remark this answers is not in the document]
Comment (gautam-e) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:55]: @jSchuetz88 do you know where I can get this from?

[the remark this answers is not in the document]
Comment (gautam-e) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:56]: I have made changes except for the example part

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl (hunk 1 above)
Comment (gautam-e) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:150]: @jSchuetz88  I need help here. Don't know what to do.

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl (hunk 3 above)
[the remark this answers is not in the document]
Comment (gautam-e) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:56]: @jSchuetz88 I need help here. Don't know what to do.

**Comment by gautam-e** [comment]
@jSchuetz88 I have made the changes that I can make. I would need some help for the remaining suggestions.

**Comment by agg3fe** [comment]
@gautam-e Could you please check eclipsefdn/eca step in the work flow. Looks like you have not signed it.

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+:PartQuantityEntity a samm:Entity ;
+    samm:preferredName "part quantity entity"@en ;
+    samm:description "The entity encapsulating the properties describing the quantity of a part."@en ;
+    samm:properties (:measurementUnit :quantityNumber) .
+
+:measurementUnit a samm:Property ;
+    samm:preferredName "measurement unit"@en ;
+    samm:description "Unit of measurement for the quantity of serialized objects."@en ;
+    samm:characteristic samm-c:UnitReference ;
+    samm:exampleValue "litre"^^samm:curie .
```
Comment (agg3fe) [inline io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:176]: example value should be "unit:litre" (without samm:curie), else it is not valid according to json schema.

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    samm-c:unit unit:piece .
+
+:PartCharacteristic a samm-c:Set ;
+    samm:preferredName "part characteristic"@en ;
+    samm:description "Characteristic of the part."@en ;
+    samm:dataType :PartEntity .
+
+:PartEntity a samm:Entity ;
+    samm:preferredName "part entity"@en ;
+    samm:description "Entity encapsulating the properies describing a part"@en ;
```
Comment (agg3fe) [inline io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:115]: typo `properies`

Comment (gautam-e) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl]: done.

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl (hunk 1 above)
Comment (jSchuetz88) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:150]: Sorry, i meant "no value example provided"

File: io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl (hunk 3 above)
[the remark this answers is not in the document]
Comment (jSchuetz88) [reply io.catenax.bom_as_specified/3.0.0/BomAsSpecified.ttl:56]: Sorry, I double checked. Please ignore that one. There is no need to provide an example value for a characteristic.

**Comment by agg3fe** [comment]
@gautam-e Could you please join the Weekly Monday meeting. We can discuss the Pull Request and provide approval. Let me know if you have meeting invite or not.

**Comment by gautam-e** [comment]
> @gautam-e Could you please join the Weekly Monday meeting. We can discuss the Pull Request and provide approval. Let me know if you have meeting invite or not.

@agg3fe I don't have the Monday appointment.

**Comment by jSchuetz88** [comment]
> > @gautam-e Could you please join the Weekly Monday meeting. We can discuss the Pull Request and provide approval. Let me know if you have meeting invite or not.
> 
> @agg3fe I don't have the Monday appointment.

Hi @gautam-e, sorry for the late reply. If you provide me your E-Mail-Adress, I will invide you to the meeting?

**Comment by gautam-e** [comment]
> > > @gautam-e Could you please join the Weekly Monday meeting. We can discuss the Pull Request and provide approval. Let me know if you have meeting invite or not.
> 
> > 
> 
> > @agg3fe I don't have the Monday appointment.
> 
> 
> 
> Hi @gautam-e, sorry for the late reply. If you provide me your E-Mail-Adress, I will invide you to the meeting?

@jSchuetz88 gautam.ethiraj@de.bosch.com

**Comment by gautam-e** [comment]
@jSchuetz88 I think @TorstenHirsch is the right person to approve this change.

**Comment by TorstenHirsch** [comment]
@gautam-e Hi Gautam, anything to do here for me? In general all Catena-X models should be in SAMM

**Comment by gautam-e** [comment]
@TorstenHirsch, I don't know for sure. 
The model `bom_as_specified` is still in bamm. That's what this PR attempted to correct. 
Perhaps @jSchuetz88 or @agg3fe can tell us what needs to be done next? I don't mind reopening this PR etc. but I believe there is some other process to be followed before merging to main (?).

**Pull request closed without merging**